README.md

```diff
@@ -0,0 +1,418 @@
+            "data_stream": {
+              "namespace": "default"
+            },
+            "id": "logfile-system-8ec9254c-0f1e-4166-965b-549c2818a0b0",
+            "meta": {
+              "package": {
+                "name": "system",
+                "version": "1.38.2"
+              }
+            },
+            "name": "system-1",
+            "package_policy_id": "8ec9254c-0f1e-4166-965b-549c2818a0b0",
+            "policy": {
+              "revision": 3
+            },
+            "revision": 2,
+            "streams": [
+              {
+                "data_stream": {
+                  "dataset": "system.auth",
+                  "type": "logs"
+                },
+                "exclude_files": [
+                  ".gz$"
+                ],
+                "id": "logfile-system.auth-8ec9254c-0f1e-4166-965b-549c2818a0b0",
+                "ignore_older": "72h",
+                "multiline": {
+                  "match": "after",
+                  "pattern": "^\\s"
+                },
+                "paths": [
+                  "/var/log/auth.log*",
+                  "/var/log/secure*"
+                ],
+                "processors": [
+                  {
+                    "add_locale": null
+                  }
+                ],
+                "tags": [
+                  "system-auth"
+                ]
+              },
+              {
+                "data_stream": {
+                  "dataset": "system.syslog",
+                  "type": "logs"
+                },
+                "exclude_files": [
+                  ".gz$"
+                ],
+                "id": "logfile-system.syslog-8ec9254c-0f1e-4166-965b-549c2818a0b0",
+                "ignore_older": "72h",
+                "multiline": {
+                  "match": "after",
+                  "pattern": "^\\s"
+                },
+                "paths": [
+                  "/var/log/messages*",
+                  "/var/log/syslog*",
+                  "/var/log/system*"
+                ],
+                "processors": [
+                  {
+                    "add_locale": null
+                  }
+                ]
+              }
+            ],
+            "type": "log"
+          },
+          "id": "logfile-system-8ec9254c-0f1e-4166-965b-549c2818a0b0",
+          "type": "log",
+          "name": "system-1",
+          "revision": "2",
+          "meta": {
+            "source": {
+              "package": {
+                "name": "system",
+                "version": "1.38.2"
+              }
+            },
+            "package": {
+              "source": {
+                "name": "system",
+                "version": "1.38.2"
+              },
+              "name": "system",
+              "version": "1.38.2"
+            }
+          },
+          "dataStream": {
+            "source": {
+              "namespace": "default"
+            },
+            "namespace": "default"
+          },
+          "streams": [
```

Comment:
What is the difference between .expected.units[n].config.source.streams` and `.expected.units[n].config.streams`?

Reply:
I need to add a top level explanation of the source field.

The source is the entire contents of the input section of the policy unmodified. the `.expected.units[n].config.streams` is only what matched against the proto message definition in this file which may not be the entire configuration. We only pull out the standardized/mandatory parts of it.

There is an attempt to explain this in the message but this pattern is used a lot, I will add a higher level explanation and make a specific note in this section.

https://github.com/elastic/elastic-agent-client/blob/c699c976fa3092435985dd633c1ed7807a753e74/elastic-agent-client.proto#L125-L142

Reply:
+1 to adding explanation of the `source` field and particularly its purpose, I still don't know why we need to move these large objects without using them.

Reply:
It is so new fields can be added without us needing to synchronize them with the release of the agent, they can just be parsed out of the source blob (which is essentially JSON encoded in protobuf).

I'll update the documentation.

Reply:
This should probably be a conversation elsewhere but this is the first I'm seeing an explicit discussion of the source field so I'll mention it here:

As things stand, it looks like a field at depth n will appear (n+1) times in the config -- once at its actual location, and then one additional time for the source field at each higher level. That's an extremely high overhead in size and complexity... is the payoff worth it? Could we potentially get by with just a single raw copy at the root level instead of recursively reduplicating every nested field? (Is it even possible for the different copies to differ, or is it strictly redundant?)

Reply:
The goal with the Source field was to ensure that the agent itself doesn't need to be updated if the YAML configuration changes.

At least in the case of the unit configurations this makes sense, since only a sub-set of the fields are required and the rest are unknown to the agent. For feature flags and component configuration the same principle seems to makes sense, but it's less clear if we'd ever use it this way.

The mechanism here was never intended to be efficient, and we've never audited any of this to minimize the size of the configuration.